store.go

```diff
@@ -2605,7 +2653,7 @@ func (s *store) DiffSize(from, to string) (int64, error) {
 }
 
 func (s *store) Diff(from, to string, options *DiffOptions) (io.ReadCloser, error) {
-	lstore, err := s.LayerStore()
+	lstore, err := s.RWLayerStore()
```

Comment:
I think that could be RO as we're not changing state.

Reply:
Let me check that via podman.

Reply:
Nah it does not seem to work: https://cirrus-ci.com/task/5661436064038912

Let me revert that change.

Reply:
Yeah, `Diff()` currently requires a write lock because we need to mount filesystems for the sake of drivers that use naive diff, and I expect `Changes()` does, too.  If we pushed the locking down into individual drivers, we could remove that requirement for drivers like overlay that use a native diff.  We'll probably have to do that, since we want to be able to push the contents of those layers to registries when those layers were part of a base image in a build.